### PR TITLE
`Development`: Remove participation status in client

### PR DESCRIPTION
**Pull request by JohannesStoehr**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.

### Motivation and Context

#5791

### Description

Remove the ParticipationStatus and replace it with the corresponding, often simpler conditions in the client code

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Students
- 1 Programming Exercise
- 1 Modeling/File Upload/Text Exercise
- 1 Quiz Exercise (synchronized mode)
- 1 Quiz Exercise (batch mode)

1. Log in to Artemis
2. Particpate in all exercises as student and check that everything (starting, submitting, reviewing result, ...) still works

#### Exam Mode Testing

Prerequisites:
- 1 Instructor
- 2 Students
- 1 Exam with each exercise types

1. Log in to Artemis
2. Participate in the exam as a student
3. Make sure that a test run still works
4. Make sure that the UI of all exercises in the exam mode stays unchanged. You can use the [exam mode documentation](https://docs.artemis.ase.in.tum.de/user/exam_mode/) as reference.

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2
#### Exam Mode Test
- [x] Test

### CODE REVIEW

File: src/test/javascript/spec/component/overview/exercise-details/exercise-details-student-actions.component.spec.ts [hunk 1]
```diff
@@ -157,14 +143,13 @@ describe('ExerciseDetailsStudentActionsComponent', () => {
...
 
-        expect(comp.participationStatusWrapper()).toEqual(ParticipationStatus.UNINITIALIZED);
         expect(startExerciseStub).toHaveBeenCalledOnce();
         participationSubject.next(initPart);
 
         fixture.detectChanges();
         tick();
 
-        expect(comp.participationStatusWrapper()).toEqual(ParticipationStatus.INITIALIZED);
+        expect(comp.studentParticipation?.initializationState).toEqual(InitializationState.INITIALIZED);
```
Comment (simon-hng) [inline src/test/javascript/spec/component/overview/exercise-details/exercise-details-student-actions.component.spec.ts:152]: could be `expect(comp.exercise.studentParticipations[0].initializationState).toEqual...`
but I am not sure if that is intended behaviour

**Review by TimOrtel — APPROVED** [review]
I looked through it and made the appropriate changes in the Android application. Thank you for your work on this. It is now a lot better than before. :rocket:

**Review by simon-hng — APPROVED** [review]
Code changes look good

File: src/main/webapp/app/overview/exercise-details/course-exercise-details.component.ts [hunk 2]
```diff
@@ -296,6 +295,7 @@ export class CourseExerciseDetailsComponent implements OnInit, OnDestroy {
             }
         }
         this.participationUpdateListener = this.participationWebsocketService.subscribeForParticipationChanges().subscribe((changedParticipation: StudentParticipation) => {
+            console.log('Update websocket');
```
Comment (Strohgelaender) [inline src/main/webapp/app/overview/exercise-details/course-exercise-details.component.ts:298]:
```suggestion
```

**Review by Strohgelaender — COMMENTED** [review]
code looks good to me.

**Review by simon-hng — APPROVED** [review]
Code

**Review by PhilipWerz — APPROVED** [review]
Tests normal exercises:
Everything worked as it should (start exercises, submitting, reviewing, complaining etc.)

Tests Exam Mode:
Everything worked and looked (UI) the same (as expected). Did same tests as for normal exercises + exam related actions 

Tested on TS1

**Review by Strohgelaender — APPROVED** [review]

**Comment by krusche** [comment]
This PR should have a few more reviews from testers to make sure that nothing is broken!

**Review by Vinicius-Agreste — APPROVED** [review]
Tested on ts1. Everything worked as expected.

**Pull request merged**